RealmSwift/Realm.swift

```diff
@@ -1231,13 +1231,17 @@ extension Realm {
         let task = RLMAsyncOpenTask(configuration: rlmConfiguration, confinedTo: scheduler,
                                     download: shouldAsyncOpen(configuration, downloadBeforeOpen))
         do {
-            try await withTaskCancellationHandler {
-                // Work around https://github.com/apple/swift/issues/61119 by smuggling
-                // the Realm out via a property on task rather than returning it
-                task.localRealm = try await task.waitForOpen()
-            } onCancel: {
-                task.cancel()
+            // Work around https://github.com/apple/swift/issues/61119, which
```

Comment:
Should this be fixed in the future , Should we mark this as FIXME or other marker?, same as the one in TestUtils